docs/differencesfromclojure.rst

```diff
@@ -32,6 +32,23 @@ Type Differences
 
   * Sorted sets, sorted maps, and array maps are not implemented (support is tracked in `#416 <https://github.com/basilisp-lang/basilisp/issues/416>`_).
 
+.. _arithmetic_comparison:
+
+Arithmetic Comparison
+---------------------
+
+Basilisp, in contrast to Clojure, does not distinguish between integer (``int``) and floating point (``float``) as `separate categories for equality comparison purposes <https://clojure.org/guides/equality>`_ where the ``=`` comparison between any ``int`` and ``float`` returns ``false``. Instead, it adopts Python's ``=`` comparison operator semantics, where the ``int`` is optimistically converted to a ``float`` before the comparison. However, beware that this conversion can lead to `certain caveats in comparison <https://stackoverflow.com/a/30100743>`_ where in rare cases seemingly exact ``int`` and ``float`` numbers may still compare to ``false`` due to limitations in floating point number representation.
+
+In Clojure, this optimistic equality comparison is performed by the ``==`` function. In Basilisp, ``==`` is aliased to behave the same as ``=``.
+
+.. note::
+
+Basilisp's ``=`` will perform as expected when using Python `Decimal <https://docs.python.org/3/library/decimal.html>`_ typed :ref:`floating_point_numbers`.
```

Comment:
```suggestion
   Basilisp's ``=`` will perform as expected when using Python `Decimal <https://docs.python.org/3/library/decimal.html>`_ typed :ref:`floating_point_numbers`.
```

I _think_ this needs to be indented to display properly.

Reply:
Right, and I was wondering while testing whether there should be a a formatting  effect on that line.

Please also note, that I had also to add a double `__` at the end of the decimal hyperlink to make it "[anonymous references](https://github.com/sphinx-doc/sphinx/issues/3921)", because it is also defined as such in the Reader section, and Sphinx was complaining with

```
docs\differencesfromclojure.rst:4: WARNING: Duplicate explicit target name: "decimal"
```

Reply:
I didn't realize that it indexed all links by title across the entire set of documents. I might need to figure out a solution for that later. Thanks for addressing that warning though.